ow_power_system/src/power_system_node.cpp

```diff
@@ -14,91 +14,131 @@
 #include <std_msgs/Int16.h>
 #include <std_msgs/Float64.h>
 
-#include "power_system_util.h"
-
+#include "power_system_node.h"
 using namespace std;
-
-int main(int argc, char* argv[]) {
-  using namespace std::chrono;
-  ros::init(argc,argv,"power_system_node");
-  ros::NodeHandle nh ("power_system_node");
-
-  //Construct our State of Charge (SOC) publisher
-  ros::Publisher SOC_pub = nh.advertise<std_msgs::Float64>("state_of_charge",1000);
-  //Construct our Remaining Useful Life (RUL) publisher (Seconds)
-  ros::Publisher RUL_pub = nh.advertise<std_msgs::Int16>("remaining_useful_life",1000);
-  //Construct our Battery Temperature (TempBat) publisher
-  ros::Publisher TempBat_pub = nh.advertise<std_msgs::Float64>("battery_temperature",1000);
-  
-  //Load power values csv
-  string csv_path;
-  string default_csv = "/data/data_const_load.csv";
-  bool csv_path_param_exist = nh.param("power_draw_csv_path", csv_path,
-    ros::package::getPath("ow_power_system") + default_csv);
-
-  if (!csv_path_param_exist)
+using namespace chrono;
+
+double update_count = 0.00; // Timestep initialization
+double battery_lifetime = 2738.0; // Estimate of battery lifetime (seconds)
+Datum<double> init_power = 0.00; // Power initialization
+Datum<double> init_temp = 20.00; // Temperature initialization
+Datum<double> init_voltage = 4.10; // Voltage initialization
+auto init_time = system_clock::now(); // Timestep initialization
+
+class SubscribeAndPublish
```

Comment:
Use a more descriptive name of the class .. for example **PowerSystemNode**..

Reply:
Resolved in latest commit.